tools/circle_plus_gen/main.py

```diff
@@ -23,19 +23,17 @@ def get_cmd_args():
     return args
 
 
-def print_training_hparameters(in_circle_file) -> typing.NoReturn:
+def print_training_hparameters(circle_model: CirclePlus):
```

Comment:
(optional) @zetwhite Why don't you define `__str__` in `CirclePlus`?

Reply:
> @zetwhite Why don't you define __str__ in CirclePlus? 

This function prints hparameter in JSON format. 
Since string format and JSON format can be different, I prefer to distinguish it by using explicit function naming.